image_segmentation/pytorch/capture_profile.py

```diff
@@ -0,0 +1,129 @@
+    )
+    parser.add_argument(
+        "--logdir",
+        dest="logdir",
+        type=str,
+        required=True,
+        help='the path to write profiling output to. Both the profiler client and server must have access. ex. "gs://bucket/file/path".',
+    )
+    parser.add_argument(
+        "--duration_ms",
+        dest="duration_ms",
+        type=int,
+        default=10000,
+        help="duration in milliseconds for tracing the server.",
+    )
+    parser.add_argument(
+        "--start_time",
+        dest="start_time",
+        type=float,
+        default=None,
+        help=(
+            "the number of seconds to sleep before starting the first profiling. "
+            "This could be a floating point number for subsecond precision. "
+            "Defaults to None, which skips sleeping."
+        ),
+    )
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument(
+        "--interactive",
+        dest="interactive",
+        type=str,
+        choices=[None, "once", "loop"],
+        default=None,
+        help=(
+            "run in interactive mode.\n"
+            'If set to "once", the profiler client asks for user confirmation before starting profiling.\n'
+            'If set to "loop", the profiler client repeatedly runs profiling, asking for user confirmation on each run.\n'
+            "Defaults to None, which disables interactive mode."
+        ),
+    )
+
+    def required_length(length):
+        class RequiredLength(argparse.Action):
+            def __call__(self, parser, args, values, option_string=None):
+                if len(values) != length:
+                    msg = f"Argument {self.dest} requires {length} arguments"
+                    raise argparse.ArgumentTypeError(msg)
+                setattr(args, self.dest, values)
+
+        return RequiredLength
+
+    group.add_argument(
+        "--automatic",
+        dest="automatic",
+        type=int,
+        nargs="+",
+        default=None,
+        action=required_length(2),
+        help=(
+            "run in automatic mode.\n"
+            "Requires 2 int type arguments.\n"
+            "The 1st argument specifies the number of profiles to capture.\n"
+            "The 2nd argument specifies the time gap (in seconds) between the profiles, "
+            "i.e. the next profiling will start X seconds after the previous profiling ends.\n"
+            'ex. "--automatic 100 60" captures 100 profiles every 60 seconds.\n'
+            "Defaults to None, which disables automatic mode."
+        ),
+    )
+
+    return parser.parse_args()
+
+
+def main():
+    args = parse_args()
+
+    def trace():
+        xp.trace(
+            service_addr=args.service_addr,
+            logdir=args.logdir,
+            duration_ms=args.duration_ms,
+        )
+        print(f"Saved profiling output to {args.logdir}")
+
+    def request_user_confirmation():
+        input("Press enter to start profiling:")
```

Comment:
nit:

Can we rephrase this statement to say: "Press enter to start profiling. Press "q" to exist profiling". 

Return "q" to break out of the infinite loop.

Reply:
Sure.

I have added an exit option to the user prompt in `capture_profile.py`
```python
def request_user_confirmation():
    usr_input = input('Press "Enter" to start profiling / Press "q" to exit profiling:')
    usr_input = usr_input.strip().lower()
    if usr_input == "q" or usr_input == "quit":
        print("Exiting gracefully...")
        sys.exit()
    elif usr_input:
        raise ValueError(f"Unknown user input: {usr_input}")
```